File: java/core/src/main/java/com/github/fernthedev/lightchat/core/data/HashedPassword.java
package com.github.fernthedev.lightchat.core.data;

import com.github.fernthedev.lightchat.core.encryption.util.EncryptionUtil;
import lombok.Getter;

import java.io.Serializable;

@Getter
public class HashedPassword implements Serializable {

    private String password;

    protected HashedPassword() {}

    /**
     * Encrypts the string using 256 hash
     * @param password
     */
    public HashedPassword(String password) {
        this.password = EncryptionUtil.makeSHA256Hash(password);
    }

    /**
     * Does not encrypt hash
     * @param hash
     * @return
     */
    public static HashedPassword fromHash(String hash) {
        HashedPassword hashedPassword = new HashedPassword();

        hashedPassword.password = hash;

        return hashedPassword;
    }
}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/data/HashedPassword.kt
package com.github.fernthedev.lightchat.core.data

import com.github.fernthedev.lightchat.core.encryption.util.EncryptionUtil
import com.squareup.moshi.JsonClass
import java.io.Serializable

@JsonClass(generateAdapter = true)
class HashedPassword
/**
 * Encrypts the string using 256 hash
 * @param password
 */
    (
    password: String,
    hash: Boolean = true
) : Serializable {
    val password = if (hash) EncryptionUtil.makeSHA256Hash(password) else password


    companion object {
        /**
         * Does not encrypt hash
         * @param hash
         * @return
         */
        fun fromHash(hash: String): HashedPassword {
            return HashedPassword(hash, false)
        }
    }
}